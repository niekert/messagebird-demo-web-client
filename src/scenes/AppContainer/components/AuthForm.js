import React from 'react';
import { func, string } from 'prop-types';
import { H1 } from 'style/Headers';
import { ContentWrapper } from 'style/Layout';
import { SubmitButton, Form, Input, ErrorLabel } from 'style/Form';
import styled from 'styled-components';

const StyledForm = styled(Form)`
  max-width: 100%;
  width: 500px;
  margin: 0 auto;
`;

class AuthForm extends React.Component {
  static propTypes = {
    authenticate: func.isRequired,
    errorMessage: string,
  };

  static defaultProps = {
    errorMessage: '',
  };

  state = {
    apiKey: '',
  };

  onInputChange = e => {
    this.setState({ apiKey: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    this.props.authenticate(this.state.apiKey);
  };

  render() {
    const { apiKey } = this.state;
    const { errorMessage } = this.props;

    return (
      <ContentWrapper>
        <StyledForm onSubmit={this.onSubmit}>
          <H1>Enter the API key to gain access</H1>
          <Input
            type="text"
            value={apiKey}
            placeholder="It's in the mail I sent you if you're supposed to be here ðŸ‘€"
            onChange={this.onInputChange}
          />
          <SubmitButton>Submit</SubmitButton>
          {!!errorMessage && <ErrorLabel>{errorMessage}</ErrorLabel>}
        </StyledForm>
      </ContentWrapper>
    );
  }
}

export default AuthForm;
