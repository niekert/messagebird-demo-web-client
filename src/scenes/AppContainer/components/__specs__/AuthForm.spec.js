import React from 'react';
import { shallow } from 'enzyme';
import AuthForm from '../AuthForm';

describe('<AuthForm />', () => {
  let props;
  beforeEach(() => {
    props = {
      authenticate: jest.fn(),
      errorMessage: '',
    };
  });

  test('renders correctly', () => {
    const wrapper = shallow(<AuthForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('submits the form when it is filled', () => {
    const wrapper = shallow(<AuthForm {...props} />);
    const mockApiKey = 'so_secure_daym';

    wrapper.setState({ apiKey: mockApiKey });
    wrapper.instance().onSubmit({ preventDefault: () => {} });

    expect(props.authenticate).toHaveBeenCalledTimes(1);
    expect(props.authenticate).toHaveBeenLastCalledWith(mockApiKey);
  });
});
