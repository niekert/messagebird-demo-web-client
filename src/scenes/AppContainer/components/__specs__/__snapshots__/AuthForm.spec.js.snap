// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> renders correctly 1`] = `
<Layout__ContentWrapper>
  <AuthForm__StyledForm
    onSubmit={[Function]}
  >
    <Headers__H1>
      Enter the API key to gain access
    </Headers__H1>
    <Form__Input
      onChange={[Function]}
      placeholder="It's in the mail I sent you if you're supposed to be here ðŸ‘€"
      type="text"
      value=""
    />
    <Form__SubmitButton>
      Submit
    </Form__SubmitButton>
  </AuthForm__StyledForm>
</Layout__ContentWrapper>
`;
